### ISSUE TITLE
Baritone hard crashes when trying to render XZ goal beacon

### ISSUE THREAD

**Issue by andre4ik3** [cabaletta/baritone#3453]
## Some information

Operating system: macOS Big Sur 11.6.5 (Apple Silicon)
Java version: openjdk 17.0.3 arm64
Minecraft version: 1.18.2 (running natively)
Baritone version: 1.8.3
Other mods (if used): None (i have some but i removed all and it still happens)

## Exception, error or logs

Here, I ran it with unoptimized
```
[20:21:40] [Render thread/INFO]: [CHAT] [Baritone] > goto 0 0
[20:21:40] [Render thread/INFO]: [CHAT] [Baritone] Going to: GoalXZ{x=0,z=0}
FATAL ERROR in native method: Thread[Render thread,10,main]: No context is current or a function that is not available in the current context was called. The JVM will abort execution.
	at org.lwjgl.opengl.GL11.glPushAttrib(Native Method)
	at baritone.utils.PathRenderer.drawDankLitGoalBox(PathRenderer.java:258)
	at baritone.utils.PathRenderer.render(PathRenderer.java:99)
	at baritone.behavior.PathingBehavior.onRenderPass(PathingBehavior.java:565)
	at baritone.event.GameEventHandler.lambda$onRenderPass$6(GameEventHandler.java:107)
	at baritone.event.GameEventHandler$$Lambda$6588/0x0000000800e94aa8.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(java.base@17.0.3/CopyOnWriteArrayList.java:807)
	at baritone.event.GameEventHandler.onRenderPass(GameEventHandler.java:107)
	- snip (obfuscated net.minecraft.client stuff) -
	at net.minecraft.client.main.Main.main(Main.java:236)
	at java.lang.invoke.LambdaForm$DMH/0x00000008002d0000.invokeStaticInit(java.base@17.0.3/LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x00000008002d0400.invokeExact_MT(java.base@17.0.3/LambdaForm$MH)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:460)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@17.0.3/Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@17.0.3/NativeMethodAccessorImpl.java:77)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@17.0.3/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@17.0.3/Method.java:568)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
```
## How to reproduce
0. have a m1 macbook
1. have minecraft running *natively* (e.g. ManyMC launcher)
2. set `renderGoalXZBeacon` to true
3. in baritone, do any command that renders a vertical goal box (e.g. goto X Z, explore)
4. observe instant minecraft hard crash

## Modified settings
`renderGoalXZBeacon` (crashes when true)

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have included the version of Minecraft I'm running, baritone's version and forge mods (if used).
- [x] I have included logs, exceptions and / or steps to reproduce the issue.
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by Mathrandom7910** [comment]
Attach the full log. And use literally any other os.

**Comment by ZacSharp** [comment]
I'd blame it on the *native* part. What ManyMc does is using custom built lwjgl (and possibly others) binaries for arm64 and the exception comes from lwjgl so there's potential for a causal relationship.

**Comment by andre4ik3** [comment]
Here is the full log. There is no "crash report" as the JVM aborts immediately without running anything else. (ignore disabled mods, they were not loaded)

Full order of things I did:
1. Load up Minecraft
2. Open Singleplayer world "New World (1)"
3. tab out for a second
4. set `renderGoalXZBeacon` to true
5. tab out again
6. run `#explore` which immediately crashed the JVM and Minecraft

[log.txt](https://github.com/cabaletta/baritone/files/8695299/log.txt)

**Comment by andre4ik3** [comment]
Not sure why you needed the full log as there is quite literally no other context than what I've already put lol.

> Attach the full log. And use literally any other os.

M1 gets 186 fps when running natively (12 chunks + sodium) and macOS is good for development, many of the tools I have (Nova, Transmit, Xcode, Suspicious Package) simply don't exist on other OSs. But I presume you mean architecture, not OS, as I don't think running it in Windows VM would solve this.

> I'd blame it on the _native_ part. What ManyMc does is using custom built lwjgl (and possibly others) binaries for arm64 and the exception comes from lwjgl so there's potential for a causal relationship.

Yeah, I was reading the logs and saw the same thing. However, either they built _release_ 3.2.2 (not some random commit on master) or downloaded some official build (lwjgl > 3 supports arm64), so I don't see why this would be an issue. I'm suspecting that running this under the normal launcher (that is, with Rosetta 2 emulation) would solve the issue, albeit with significantly poorer performance (like 30 fps).

Maybe there is some way to at least prevent the crash. Because here it doesn't even get to generate the crash report lol.

**Comment by scorbett123** [comment]
The issue you've got is that the many mc opengl mappings don't seem to support glPushAttrib, there isn't much we can do about this, so really you can only run minecraft non-natively.

**Comment by andre4ik3** [comment]
Isn't `glPushAttrib` [deprecated](https://community.khronos.org/t/glpushattrib-depreciated-replace-by/63944)?

**Comment by scorbett123** [comment]
yea, I think it might be, was actually just researching some stuff about it. At the moment on normal mc it works, so whilst it still works on normal mc this is most likely be a low priority, however if anyone want to try and change the glPushAttrib then that pr will probably be accepted.

If you rename this to "Stop using glPushAttrib as it is deprecated" then I'll rename this.

**Comment by andre4ik3** [comment]
Alright

**Comment by andre4ik3** [comment]
From what I've read and understand, glPushAttrib is basically just a wrapper that does some management stuff for you. Maybe this is a good opportunity for me to learn Java & OpenGL and make a PR :)

**Comment by ZeroMemes** [comment]
Fixed by #4059

**Comment by ZacSharp** [comment]
Still not fixed on master

**Comment by ZeroMemes** [comment]
Fully resolved by 90f2b39, although vanilla code in 1.12.2 still uses glPushAttrib and glPopAttrib.

**Comment by ZacSharp** [comment]
Thanks for fixing it anyways (I like cross version consistency).
